data/inference/run_api.py

```diff
@@ -0,0 +1,159 @@
+import json
+import os
+import traceback
+from pathlib import Path
+
+import dotenv
+import fire
+import numpy as np
+import tiktoken
+from datasets import load_dataset, load_from_disk
+from make_datasets.utils import extract_diff
+from tenacity import retry, stop_after_attempt, wait_random_exponential
+from tqdm.auto import tqdm
+
+from metagpt.config2 import config
+from metagpt.logs import logger
+from metagpt.roles.di.data_interpreter import DataInterpreter
+from metagpt.utils.recovery_util import save_history
+
+dotenv.load_dotenv()
+
+# Replace with your own
+MAX_TOKEN = 128000
+
+
+@retry(wait=wait_random_exponential(min=30, max=600), stop=stop_after_attempt(5))
+async def call_chat(inputs, interpreter):
+    """
+    Calls the openai API to generate completions for the given inputs.
+
+    Args:
+    inputs (str): The inputs to generate completions for.
+    interpreter (DataInterpreter): The data interpreter to use for execution.
+    """
+    system_messages = inputs.split("\n", 1)[0]
+    user_message = inputs.split("\n", 1)[1]
+    try:
+        await interpreter.run([system_messages, user_message])
+        code = interpreter.execute_code.nb.cells[-1].source
+        return code
+    except Exception as e:
+        logger.error(e)
+        logger.error(f"Inputs: {inputs}")
+        traceback.print_exc()
+        raise e
```

Comment:
1. I think most of this function should be written inside the interpreter
2. The two lines of logger.error should be reduced to one line